geoindicators/src/main/groovy/org/orbisgis/RsuIndicators.groovy

```diff
@@ -619,4 +615,164 @@ static IProcess rsuEffectiveTerrainRoughnessHeight() {
+                    // If the road direction distribution is calculated, explode the roads into segments in order to calculate
+                    // their length for each azimuth range
+                    if (operations.contains("rsu_road_direction_distribution")) {
+                        def queryExpl = "DROP TABLE IF EXISTS $roadExpl;" +
+                                "CREATE TABLE $roadExpl AS SELECT id_rsu, the_geom, " +
+                                "CASEWHEN(ST_AZIMUTH(ST_STARTPOINT(the_geom), ST_ENDPOINT(the_geom))>=pi()," +
+                                "ROUND(DEGREES(ST_AZIMUTH(ST_STARTPOINT(the_geom), ST_ENDPOINT(the_geom))))-180," +
+                                "ROUND(DEGREES(ST_AZIMUTH(ST_STARTPOINT(the_geom), ST_ENDPOINT(the_geom))))) AS azimuth,"+
+                                "ST_LENGTH(the_geom) AS length $ifZindex FROM ST_EXPLODE('(SELECT ST_TOMULTISEGMENTS(the_geom)" +
+                                " AS the_geom, id_rsu $ifZindex FROM $roadInter)');"
+                        // Calculate the road direction for each direction and optionally level
+                        def queryDistrib = queryExpl+"CREATE TABLE $roadDistrib AS SELECT id_rsu, "+caseQueryDistrib[0..-2]+
+                                " FROM $roadExpl GROUP BY id_rsu;"+
+                                "CREATE INDEX IF NOT EXISTS id_u ON $rsuTable($idColumnRsu);"+
+                                "CREATE INDEX IF NOT EXISTS id_d ON $roadDistrib(id_rsu);"+
+                                "DROP TABLE IF EXISTS $roadDistTot; CREATE TABLE $roadDistTot AS SELECT b.* " +
+                                "FROM $rsuTable a LEFT JOIN $roadDistrib b ON a.$idColumnRsu=b.id_rsu;"
+                        datasource.execute(queryDistrib.toString())
+                        if(operations.contains("rsu_linear_road_density") == false){
+                            datasource.execute("ALTER TABLE $roadDistTot RENAME TO $outputTableName".toString())
+                        }
+                    }
+                    // If the rsu linear density should be calculated
+                    if(operations.contains("rsu_linear_road_density")){
+                        String queryDensity = "CREATE TABLE $roadDens AS SELECT id_rsu, "+caseQueryDens[0..-2]+
+                                " FROM $roadInter GROUP BY id_rsu;"+
+                                "CREATE INDEX IF NOT EXISTS id_u ON $rsuTable($idColumnRsu);"+
+                                "CREATE INDEX IF NOT EXISTS id_d ON $roadDens(id_rsu);"+
+                                "DROP TABLE IF EXISTS $roadDensTot; CREATE TABLE $roadDensTot AS SELECT b.* " +
+                                "FROM $rsuTable a LEFT JOIN $roadDens b ON a.$idColumnRsu=b.id_rsu"
+                        datasource.execute(queryDensity.toString())
+                        if(operations.contains("rsu_road_direction_distribution") == false){
```

Comment:
use
!operations.contains("rsu_road_direction_distribution")

Reply:
I have made the modifications for all occurences